plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-custom-fields-controller.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * REST API CustomFields controller
+ *
+ * Handles requests to the /products endpoint.
+ *
+ * @package WooCommerce\RestApi
+ * @since   9.2.0
+ */
+
+use Automattic\WooCommerce\Utilities\I18nUtil;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST API Product Custom Fields controller class.
+ *
+ * @package WooCommerce\RestApi
+ * @extends WC_REST_Controller
+ */
+class WC_REST_Product_Custom_Fields_Controller extends WC_REST_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'products/custom-fields';
+
+	/**
+	 * Post type.
+	 *
+	 * @var string
+	 */
+	protected $post_type = 'product';
+
+	/**
+	 * Register the routes for products.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/names',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item_names' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Get a collection of custom field names.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_item_names( $request ) {
+		global $wpdb;
+
+		$search = trim( $request['search'] );
+		$order  = strtoupper( $request['order'] ) === 'DESC' ? 'DESC' : 'ASC';
+		$page   = (int) $request['page'];
+		$limit  = (int) $request['per_page'];
+		$offset = ( $page - 1 ) * $limit;
+
+		$base_query = $wpdb->prepare(
+			"SELECT DISTINCT m.meta_key
```

Comment:
Totally nitpicky and optional. We usually use more descriptive names for the aliases in our SQL queries, and it would be nice to keep that. As I mentioned is not a big deal.

Reply:
done -> [7978932](https://github.com/woocommerce/woocommerce/pull/48504/commits/79789323f8377385fe1b09aa25f7ed4d10ee0aea)